renaissance/scripts/perf/getmetrics-promql.sh

```diff
@@ -0,0 +1,271 @@
+#!/bin/bash
+#
+# Copyright (c) 2020, 2021,2022 IBM Corporation, RedHat and others.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+### Script to get pod and cluster information through prometheus queries###
+#
+# checks if the previous command is executed successfully
+# input:Return value of previous command
+# output:Prompts the error message if the return value is not zero
+function err_exit() 
+{
+	if [ $? != 0 ]; then
+		printf "$*"
+		echo 
+		exit 1
+	fi
+}
+function get_cpu()
+{
+	URL=$1
+	TOKEN=$2
+	RESULTS_DIR=$3
+	ITER=$4
+	APP_NAME=$5
+	# Delete the old json file if any
+	rm -rf ${RESULTS_DIR}/cpu-${ITER}.json
+	while true
+	do
+		# Processing curl output "timestamp value" using jq tool.
+#		echo "curl --silent -G -kH Authorization: Bearer ${TOKEN} --data-urlencode 'query=sum(node_namespace_pod_container:container_cpu_usage_seconds_total:sum_rate) by (pod)' ${URL} "		 
+	  curl --data-urlencode 'query=sum(rate(container_cpu_usage_seconds_total[5m])) by (pod,namespace)' http://localhost:9090/api/v1/query | jq '[ .data.result[] |  [.metric.pod, .value[0], .value[1]|tostring]| join(";") ]' | grep "${APP_NAME}"| cut -d ";" -f2,3 >> ${RESULTS_DIR}/cpu-${ITER}.json
```

Comment:
Define "http://localhost:9090/api/v1/query" as a variable ( something like "URL") and use for all instances. URL may change based on the environment it is running on.

Reply:
Include --silent in the curl command